src/FESpaces/Assemblers.jl

```diff
@@ -270,44 +270,88 @@ function assemble_matrix(f::Function,a::Assembler,U::FESpace,V::FESpace)
   assemble_matrix(a,collect_cell_matrix(U,V,f(u,v)))
 end
 
+function assemble_matrix!(A,f::Function,a::Assembler,U::FESpace,V::FESpace)
```

Comment:
Start the argument list with ::Function objects so that we can use do-block syntax. Also type-annotate the new array arguments, I.e.
```julia
assemble_matrix!(f::Function,A::AbstractMatrix,a::Assembler,U::FESpace,V::FESpace)
assemble_matrix_and_vector!(f::Function,b::Function,M::AbstractMatrix,r::AbstractVector,a::Assembler,U::FESpace,V::FESpace)
```
Apply this minor changes to all new introduced assembly functions.

Reply:
> Start the argument list with ::Function objects so that we can use do-block syntax

I though that the parameters that are modified in a `!`-like function should appear first in the list of parameters by convention. From your comment, I guess this is not strictly necessary, right?

Reply:
right, see e.g. `broadcast`